Updates for latest Open J9 with checkpoint support

tjwatson opened the pull request:
Open J9 now supports immutable environment variables.  All other environment variables will return null until a restore.

tjwatson committed:
Add a CheckpointPhase.restored method

The restored() method returns true if the running process is a
checkpoint restored process.

Also added a CheckpointPhase service property
io.openliberty.checkpoint.restored that gets set to true on restore.

The restored service property is updated after enabling multi-threaded
mode to avoid firing modified service events on restore while in
single-threaded mode.

tjwatson committed:
Update Liberty to handle OpenJ9 env variable support on restore

tjwatson committed:
Add tests to verify restored method and service property on CheckpointPhase

tjwatson committed:
Enable the envValueChangeTest

This test can now be enabled with the OpenJ9 support for immutable vs
mutable environment variables.

tjwatson commented:
#build

sebratton commented on dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:
```diff
@@ -919,7 +922,16 @@ javaCmd()
+    X_WLP_IMMUTABLE_VARS="-Dorg.eclipse.openj9.criu.ImmutableEnvVars=INVOKED,WLP_USER_DIR,WLP_OUTPUT_DIR,LOG_DIR,X_LOG_DIR,LOG_FILE,X_LOG_FILE,VARIABLE_SOURCE_DIRS,X_CMD"
+
+    # Add any additional user specified variables
+    if [ -z "$WLP_IMMUTABLE_VARS" ]; then
```
should this be 
    [ -n "$WLP_IMMUTABLE_VARS"]

tjwatson replied:
Done

sebratton commented on dev/com.ibm.ws.kernel.boot.ws-server/publish/bin/server:
```diff
@@ -1363,11 +1375,8 @@ restoreServer()
-  # - _ holds the currently running process, here we do no need to have it set to `env`
-  env | grep -Ev "JAVA_OPTIONS=|SHLVL=|HOSTNAME=|_=|X_LOG_FILE=|X_LOG_DIR=" > ${SERVER_OUTPUT_DIR}/workarea/checkpoint/.env.properties
+  # - INVOKED - this gets set to the path used to launch the server script, we don't care if this changes
+  env | grep -Ev "INVOKED=" > ${SERVER_OUTPUT_DIR}/workarea/checkpoint/.env.properties
```
Since this filters out any var ending in INVOKED should it be more specific like:
  env | grep -Ev "^INVOKED="

tjwatson replied:
Done

tjwatson committed:
Correctly check for WLP_IMMUTABLE_VARS and INVOKED

Add a testcase for WLP_IMMUTABLE_VARS to make sure
additional immutable env vars can be set.

tjwatson commented:
#build

tjwatson commented:
#build

tjwatson commented:
#libby
